src/Syntax/TextmateTokenizerJob.re

```diff
@@ -100,6 +101,20 @@ let onBufferUpdate = (bufferUpdate: BufferUpdate.t, lines, v: t) => {
   Job.map(f, v);
 };
 
+let _hexToColor = Hashtbl.create(128);
```

Comment:
A leading underscore is still not a good idea as it has a different compiler-enforced meaning than you want it to have. I'd suggest using an `Internal` module instead, or if it's just used for a single function, scope it to that. This could even be generalized into a `cached` utility function I think:

```
let cached = (size, f) => {
  let cache = Hashtbl.create(size);

  input =>
    switch (Hashtbl.find_opt(cache, input)) {
    | Some(output) => output
    | None =>
      let output = f(input);
      Hashtbl.add(cache, input, output);
      output;
    };
};
```